crates/miden-objects/src/asset/token_symbol.rs

```diff
@@ -1,20 +1,23 @@
-use alloc::string::String;
+use alloc::string::{String, ToString};
 
-use super::{AssetError, Felt};
+use super::{Felt, TokenSymbolError};
 
 #[derive(Default, Clone, Copy, Debug)]
 pub struct TokenSymbol(Felt);
```

Comment:
While we're improving this type, let's also add doc comments to this struct, all constants and all functions.